### PR TITLE
Plugin MVP

### PR DESCRIPTION
**Pull request by mboynes**
This is a plugin that allows one WordPress site to handle multiple domains. In other words, you can have multiple user-facing websites run by one (non-multisite) admin. This PR is the MVP for the plugin.

### CODE REVIEW

File: lib/class-core.php [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+					return;
+				}
+			}
+
+			$current_site = $this->get_current_site_term();
+			if ( ! $current_site || $post_site->name === $current_site->name ) {
+				return;
+			}
+
+			wp_redirect( str_replace( $current_site->name, $post_site->name, get_permalink() ), 301 );
```
Comment (bcampeau) [inline lib/class-core.php:246]: You consider using `wp_safe_redirect` so you could pull from a list of approved hosts via the plugin? Just a thought, VIP might like that.

File: lib/class-core.php [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+			|| ! isset( $_SERVER['REQUEST_URI'] )
+			|| defined( 'DOING_AJAX' ) && DOING_AJAX
+			|| ! apply_filters( 'split_domain_redirect_admin_domain', true )
+		) {
+			return;
+		}
+
+		$site = $this->get_default_site();
+		if ( ! empty( $site->name ) && false === strpos( $_SERVER['HTTP_HOST'], $site->name ) ) { // WPCS: sanitization ok.
+			wp_redirect( esc_url_raw( 'http://' . $site->name . $_SERVER['REQUEST_URI'] ) ); // WPCS: sanitization ok.
```
Comment (bcampeau) [inline lib/class-core.php:325]: See above

File: lib/class-core.php [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+	 *     @type string $slug The site's term slug.
+	 * }
+	 */
+	public static function get_current_site_from_cache() {
+		$domains = get_option( 'split_domain_sites', [] );
+		if ( empty( $domains ) ) {
+			return;
+		}
+
+		$current_host = parse_url( home_url(), PHP_URL_HOST );
```
Comment (nschoenholtz) [inline lib/class-core.php:102]: Doesn't this need to be `$_SERVER['HTTP_HOST']`?

Comment (mboynes) [reply lib/class-core.php:102]: not if the `WP_HOME` constant is set

File: lib/class-core.php (hunk 1 above)
Comment (mboynes) [reply lib/class-core.php:246]: This is a good idea. I'm logging it as an issue to implement later.

**Pull request merged**